File: app/src/main/java/com/cyberegylet/antiDupeGallery/backend/Activites/ActivityParameter.java
package com.cyberegylet.antiDupeGallery.backend.Activites;

import android.net.Uri;

public final class ActivityParameter
{
	public final String name;
	public final Object data;
	public final Type type;

	public enum Type
	{
		INT,
		STRING,
		BOOL,
		URI,
	}

	public ActivityParameter(String name, boolean data)
	{
		this.name = name;
		this.data = data;
		this.type = Type.BOOL;
	}

	public ActivityParameter(String name, int data)
	{
		this.name = name;
		this.data = data;
		this.type = Type.INT;
	}

	public ActivityParameter(String name, String data)
	{
		this.name = name;
		this.data = data;
		this.type = Type.STRING;
	}

	public ActivityParameter(String name, Uri data)
	{
		this.name = name;
		this.data = data;
		this.type = Type.URI;
	}
}

File: app/src/main/java/com/cyberegylet/antiDupeGallery/helpers/activities/ActivityParameter.kt
package com.cyberegylet.antiDupeGallery.helpers.activities

import android.net.Uri

class ActivityParameter<T : Any?> private constructor(val name: String, val data: Any, val type: Type)
{
	enum class Type
	{
		INT,
		STRING,
		STRING_ARR,
		BOOL,
		URI,
		PARCELABLE
	}

	constructor(name: String, data: Boolean) : this(name, data, Type.BOOL)
	constructor(name: String, data: Int) : this(name, data, Type.INT)
	constructor(name: String, data: String) : this(name, data, Type.STRING)
	constructor(name: String, data: Array<String>) : this(name, data, Type.STRING_ARR)
	constructor(name: String, data: Uri) : this(name, data, Type.URI)
	constructor(name: String, data: ArrayList<T>) : this(name, data, Type.PARCELABLE)
}